### PR TITLE
MMT-3771: Add Google Analytics 4 code

### PR DESCRIPTION
**Pull request by htranho**
# Overview

### What is the feature?

Add GA4 to React MMT

### What is the Solution?

Add GA4 code to the index page

### What areas of the application does this impact?

Index page

# Testing

GA4 team will check MMT events sent to GA4 

### Attachments

N/A

# Checklist

- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by cgokey — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -6,7 +6,31 @@
...
+                  new Date().getTime(),
+          event: 'gtm.js'
+        });
+        var f = d.getElementsByTagName(s)[0],
+                j = d.createElement(s),
+                dl = l != 'dataLayer' ? '&l=' + l : '';
+        j.async = true;
+        j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
+        f.parentNode.insertBefore(j, f);
+      })(window, document, 'script', 'dataLayer', 'GTM-WNP7MLF');
```
Comment (macrouch) [inline index.html:24]: I'm not sure if you want to be committing an ID like this into the repo.

And do you want to report to Google in the dev environment?

Comment (trevorlang) [reply index.html:24]: I don't think there is any harm in committing the ID but it is nice to pull those values from a config to make local development/debugging easier.

Though, I assume we probably want this set up similarly to Earthdata Search. When in development we initialize the development GA instance directly, without tag manager. And when in SIT, UAT, or Prod initialize the appropriate Tag Manager container (GTM-WNP7MLF), which will load the appropriate GA instance for the particular environment.

Comment (trevorlang) [reply index.html:24]: @htranho - Just so you're aware, I made some edits to the comment above after looking deeper at the code.

Comment (htranho) [reply index.html]: Updated: GA4 ID as passed in variable, please review.

File: static/src/js/components/OneOfField/OneOfField.jsx [hunk 2]
```diff
@@ -98,12 +101,9 @@ class OneOfField extends React.Component {
       }
     }
 
-    /*
-    *  The code below commented out caused a bug where it would revert the option selected in some cases.
-    */
-    // if (newState !== this.state) {
-    //   this.setState(newState)
-    // }
+    if (newState !== this.state) {
```
Comment (cgokey) [inline static/src/js/components/OneOfField/OneOfField.jsx:104]: Why are these changes showing up in your PR?

Comment (htranho) [reply static/src/js/components/OneOfField/OneOfField.jsx]: rebase issue. Deep fixed.

**Review by trevorlang — APPROVED** [review]

**Comment by htranho** [comment]
Put status to 'draft' because issue with rebasing from MMT-3390 (files show up as changes which have nothing to do with MMT-3771)

**Pull request merged**